File: src/lib/api/portfolio.ts
import { buildPublicApiUrl } from "@/lib/api/public";

export interface PublicPortfolioItem {
  id: string;
  service_slug: string;
  title: string;
  category: string | null;
  image_url: string;
  link_url: string | null;
  sort_order: number;
  created_at: string;
}

export async function fetchPortfolios(service_slug: string): Promise<PublicPortfolioItem[]> {
  try {
    const url = buildPublicApiUrl("portfolios");
    url.searchParams.set("service_slug", service_slug);

    const res = await fetch(url.toString(), { cache: "no-store" });
    if (!res.ok) return [];
    return res.json();
  } catch {
    return [];
  }
}

/** Ambil SEMUA porto (lintas layanan) untuk galeri /portofolio. */
export async function fetchAllPortfolios(): Promise<PublicPortfolioItem[]> {
  try {
    const url = buildPublicApiUrl("portfolios");
    const res = await fetch(url.toString(), { cache: "no-store" });
    if (!res.ok) return [];
    return res.json();
  } catch {
    return [];
  }
}

File: backend/app/schemas/portfolio.py
from pydantic import BaseModel


class PortfolioBase(BaseModel):
    service_slug: str
    title: str
    category: str | None = None
    image_url: str
    sort_order: int = 0


class PortfolioCreate(PortfolioBase):
    pass


class PortfolioUpdate(BaseModel):
    service_slug: str | None = None
    title: str | None = None
    category: str | None = None
    image_url: str | None = None
    sort_order: int | None = None


class PortfolioOut(PortfolioBase):
    id: str
    created_at: str
    model_config = {"from_attributes": True}
